lib/src/main/kotlin/com/github/bratek20/hla/generation/impl/core/api/ComplexStructureField.kt

```
@@ -58,10 +61,22 @@ open class ComplexStructureField(
         }
     }
 
+    fun buildExampleValue(): Any? {
+        val value = this.extractExampleValue()
+        if(value != null) {
+            if(type is BaseApiType) {
+                val basApiType = type as BaseApiType
+                return BaseApiType.parseToProperExampleFormat(basApiType, value)
+            }
+            return destringify(value)
```

IMO it is better to throw here ShouldNeverHappen exception - currently HLA is not supporting example values for non base types - it is better to fail fast than providing implementation that most likely still will be wrong in the future

---

I think in the best case we can change it to return value but adding here the exception is throwing error for wrapped non baseApiType like Date